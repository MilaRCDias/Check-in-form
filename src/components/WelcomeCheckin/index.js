import React from "react";
import PropTypes from "prop-types";
import { Formik, Form } from "formik";
import * as Yup from "yup";
import { TextField, Button, Box, Typography } from "@material-ui/core";
import { makeStyles, createStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() =>
  createStyles({
    error: {
      color: "red",
      fontStyle: "italic",
    },
  })
);

/**
 * Welcome checkin component
 * Step 1 of checkin form
 */
const WelcomeCheckin = ({ handleStep, currentStep }) => {
  const style = useStyles();

  /** Validation schema */
  const validation = Yup.object().shape({
    flightNumber: Yup.string()
      .min(2, "Too Short!")
      .required("Enter a valid flight number. No special character allow")
      .test({
        test: function (flightNumber) {
          const isValid = /[!@#$%^&*(),.?":{}|<>]/.test(flightNumber);
          return !isValid;
        },
      }),
    lastName: Yup.string()
      .min(2, "Too Short!")
      .max(50, "Too Long!")
      .required("Lastname is required"),
  });

  /** submit values */
  const onSubmitForm = (values) => {
      handleStep(values);
  };

  return (
    <div>
      {currentStep === 1 ? (
        <Box my={8} mx={4}>
          <Typography
            variant="h4"
            component="h1"
            align="left"
            color="textSecondary"
          >
            {" "}
            Welcome to your web check-in
          </Typography>
          <Formik
            initialValues={{ flightNumber: "", lastName: "" }}
            validationSchema={validation}
            onSubmit={onSubmitForm}
          >
            {({ errors, touched, handleChange, isSubmitting }) => {
              return (
                <Form>
                  <Box my={6}>
                    <TextField
                      id="flightNumber"
                      label="Flight Number"
                      onChange={handleChange}
                      error={!!errors.flightNumber}
                      fullWidth
                      variant="outlined"
                    />
                    {errors.flightNumber && touched.flightNumber ? (
                      <div className={style.error}>{errors.flightNumber}</div>
                    ) : null}
                  </Box>

                  <Box>
                    <TextField
                      id="lastName"
                      label="Last Name"
                      onChange={handleChange}
                      error={!!errors.lastName}
                      fullWidth
                      variant="outlined"
                    />
                    {errors.lastName && touched.lastName ? (
                      <div className={style.error}>{errors.lastName}</div>
                    ) : null}
                  </Box>
                  <br />
                  <br />
                  <Button
                    type="submit"
                    variant="contained"
                    color="primary"
                    disabled={isSubmitting}
                  >
                    Submit
                  </Button>
                </Form>
              );
            }}
          </Formik>
        </Box>
      ) : null}
    </div>
  );
};

WelcomeCheckin.propTypes = {
  /** handle submit form */
  handleStep: PropTypes.func.isRequired,
  /**the number of the form step */
  currentStep: PropTypes.number.isRequired,
};

export default WelcomeCheckin;
