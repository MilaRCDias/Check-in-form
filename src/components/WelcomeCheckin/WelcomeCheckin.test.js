import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import testConfig from "../../../testConfig";
import { mount } from "enzyme";
import WelcomeCheckin from "./index";

describe("the welcome checkin step", () => {
  let shallow;
  const props = {
    currentStep: 1,
    handleStep: () => {},
  };
  beforeAll(() => {
    shallow = createShallow();
  });

  it("should capture last name", () => {
    const credentials = { lastName: "wild name" };

    const wrapper = mount(shallow(<WelcomeCheckin />).get(0));
    const input = wrapper.find("#lastName");

    input.value = credentials.lastName;
    expect(input.value).toEqual("wild name");
  });

  it("should capture flight number", () => {
    const credentials = { flightNumber: "fr1234" };

    const wrapper = mount(shallow(<WelcomeCheckin />).get(0));
    const input = wrapper.find("#flightNumber");

    input.value = credentials.flightNumber;
    expect(input.value).toEqual("fr1234");
  });

  it("should match the snapshot", () => {
    const wrapper = mount(shallow(<WelcomeCheckin  />).get(0));

    expect(wrapper.html()).toMatchSnapshot();
  });
});
