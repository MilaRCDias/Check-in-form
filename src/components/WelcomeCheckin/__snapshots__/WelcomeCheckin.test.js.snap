// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the welcome checkin step should match the snapshot 1`] = `"<div><div class=\\"MuiBox-root MuiBox-root-12\\"><h1 class=\\"MuiTypography-root MuiTypography-h4 MuiTypography-colorTextSecondary MuiTypography-alignLeft\\"> Welcome to your web check-in</h1><form action=\\"#\\"><div class=\\"MuiBox-root MuiBox-root-13\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"flightNumber\\" id=\\"flightNumber-label\\">Flight Number</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"flightNumber\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-6\\"><span>Flight Number</span></legend></fieldset></div></div></div><div class=\\"MuiBox-root MuiBox-root-14\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"lastName\\" id=\\"lastName-label\\">Last Name</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl\\"><input aria-invalid=\\"false\\" id=\\"lastName\\" type=\\"text\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\" value=\\"\\"><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-4 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-6\\"><span>Last Name</span></legend></fieldset></div></div></div><br><br><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary\\" tabindex=\\"0\\" type=\\"submit\\"><span class=\\"MuiButton-label\\">Submit</span><span class=\\"MuiTouchRipple-root\\"></span></button></form></div></div>"`;
